code/datalab-app/src/app/components/app/TopBar.js

```diff
@@ -1,16 +1,20 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import AppBar from 'material-ui/AppBar';
 import Toolbar from 'material-ui/Toolbar';
-import Button from 'material-ui/Button';
-import auth from '../../auth/auth';
+import UserInfo from './UserIcon';
 
-const TopBar = () => (
+const TopBar = ({ identity }) => (
   <AppBar position="fixed">
     <Toolbar>
       <div style={{ flex: 1 }} />
-      <Button color="primary" raised onClick={auth.logout}>Logout</Button>
+      <UserInfo identity={identity} />
```

Comment:
Surprised this didn't break the spec. You're suffering from a new plugin I just installed in Chrome that gives me a tree view of the PR. I can't believe I haven't had this before, it will make my PRs so much better...or worse depending on how you see it. In case you're wondering, it is Octotree https://chrome.google.com/webstore/detail/octotree/bkhaagjahfmjljalopjnoealnfndnagc

Reply:
Will check it out.